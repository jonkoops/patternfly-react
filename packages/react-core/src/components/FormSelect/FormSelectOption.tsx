import { HTMLProps, FunctionComponent } from 'react';

export interface FormSelectOptionProps extends Omit<HTMLProps<HTMLOptionElement>, 'disabled'> {
  /** additional classes added to the Select Option */
  className?: string;
  /** the value for the option */
  value?: any;
  /** the label for the option */
  label: string;
  /** flag indicating if the option is disabled */
  isDisabled?: boolean;
  /** flag indicating if option will have placeholder styling applied when selected **/
  isPlaceholder?: boolean;
}

export const FormSelectOption: FunctionComponent<FormSelectOptionProps> = ({
  className = '',
  value = '',
  isDisabled = false,
  label,
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  isPlaceholder = false,
  ...props
}: FormSelectOptionProps) => (
  <option {...props} className={className} value={value} disabled={isDisabled}>
    {label}
  </option>
);
FormSelectOption.displayName = 'FormSelectOption';
