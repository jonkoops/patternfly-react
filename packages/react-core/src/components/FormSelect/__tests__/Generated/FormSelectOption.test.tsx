/**
 * @vitest-environment jsdom
 */
import { render } from '@testing-library/react';
import React from 'react';
import { expect, it } from 'vitest';

import { FormSelectOption } from '../../FormSelectOption';

it('FormSelectOption should match snapshot (auto-generated)', () => {
  const { asFragment } = render(<FormSelectOption className={"''"} value={''} label={'string'} isDisabled={false} />);
  expect(asFragment()).toMatchSnapshot();
});
