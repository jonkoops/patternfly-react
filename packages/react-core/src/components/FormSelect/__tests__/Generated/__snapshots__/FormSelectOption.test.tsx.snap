// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormSelectOption should match snapshot (auto-generated) 1`] = `
<DocumentFragment>
  <option
    class="''"
    value=""
  >
    string
  </option>
</DocumentFragment>
`;
